af_ktls.c

```diff
@@ -286,15 +286,21 @@ static inline struct tls_sock *tls_sk(struct sock *sk)
 	return (struct tls_sock *)sk;
 }
 
-void increment_seqno(unsigned char *seq)
+void increment_seqno(unsigned char *seq, struct tls_sock *tsk)
 {
-    int i;
+	int i;
 
-    for (i = 7; i >= 0; i--) {
-        ++seq[i];
-        if (seq[i] != 0)
-            break;
-    }
+	for (i = 7; i >= 0; i--) {
+		++seq[i];
+		if (seq[i] != 0)
+			break;
+	}
+	/*
+	 * Check for overflow. If overflowed, connection must either rehandshake
+	 * or disconnect. Raise an error and notify userspace
+	 */
```

Comment:
A nitpick on the comment. If the overflow occurs you can no longer rehandshake as you have no more numbers to use to exchange the handshake packets. At this point you can only terminate the session.